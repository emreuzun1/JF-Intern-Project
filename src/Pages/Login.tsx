import {StackNavigationProp} from '@react-navigation/stack';
import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  Dimensions,
  Image,
  TouchableOpacity,
  TextInput,
  Button,
} from 'react-native';
import {useDispatch, connect, useSelector} from 'react-redux';
import {RouteProp} from '@react-navigation/native';

import {RootStackParamList} from '../Navigation/types';
import {requestLogin} from '../redux/actions';
import {Formik} from 'formik';
import Waiting from '../components/Waiting';
import {AppDispatch} from '../redux/store';
import { IState } from '../Interfaces/actionInterface';

const {width} = Dimensions.get('screen');
const logoImg = require('../img/jotform-logo.png');

type LoginProps = StackNavigationProp<RootStackParamList, 'Login'>;
type LoginRouteProps = RouteProp<RootStackParamList, 'Login'>;

interface IProps {
  navigation: LoginProps;
  route: LoginRouteProps;
  appKey: '';
}

interface IFormValues {
  username: string;
  password: string;
}

interface DispatchProps {
  loginTo: (username: string, password: string, navigation : any) => void;
}

type Props = IProps & DispatchProps;

const mapStateToProps = (state: any) => {
  const appkey = state.auth;
  return appkey;
};

const mapDispatchToProps = (dispatch: any) => {
  return {
    loginTo: (username: string, password: string, navigation: any) => {
      dispatch(requestLogin(username, password));
      navigation.navigate('Form');
    },
  };
};

const Login = (props: Props) => {
  const initialValues: IFormValues = {username: '', password: ''};
  const dispatch: AppDispatch = useDispatch();
  const appkey = useSelector((state : IState)=> state.auth.appKey);

  const {appKey, navigation , route} = props;

  if (route.params.isLogged) {
    navigation.navigate('Form');
    return <Waiting />;
  }
  return (
    <View style={styles.container}>
      <View style={styles.topContainer}>
        <Image style={styles.logoImg} source={logoImg} />
      </View>
      <View style={styles.bottomContainer}>
        <Formik
          initialValues={initialValues}
          onSubmit={values => props.loginTo(values.username, values.password, navigation)}>
          {({handleChange, handleBlur, handleSubmit, values}) => (
            <View style={styles.inputContainer}>
              <TextInput
                style={styles.input}
                placeholder=" Username"
                value={values.username}
                onBlur={handleBlur('username')}
                onChangeText={handleChange('username')}
              />
              <TextInput
                style={styles.input}
                placeholder=" Password"
                value={values.password}
                secureTextEntry
                onBlur={handleBlur('password')}
                onChangeText={handleChange('password')}
              />
              <Button title="Login" onPress={handleSubmit} />
            </View>
          )}
        </Formik>
        <TouchableOpacity style={styles.signUp} onPress={() => {}}>
          <Text>
            Don't you have an account?{' '}
            <Text style={{color: '#fa8900', fontWeight: '700'}}>Sign up!</Text>
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#fa8900',
  },

  button: {
    backgroundColor: '#434343',
    width: width / 2.6,
    padding: 15,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 8,
    marginTop: 24,
  },

  topContainer: {
    flex: 2,
    justifyContent: 'center',
  },

  bottomContainer: {
    flex: 3,
    backgroundColor: 'white',
    width: '100%',
    alignItems: 'center',
    borderTopLeftRadius: 36,
    borderTopRightRadius: 36,
    paddingTop: 24,
  },

  inputContainer: {
    alignItems: 'center',
    marginTop: 12,
  },

  logoImg: {
    width: width / 1.2,
    resizeMode: 'contain',
    marginBottom: 12,
  },

  input: {
    backgroundColor: '#e3e3e3e3',
    borderRadius: 5,
    width: width / 1.6,
    marginVertical: 10,
    fontSize: 16,
  },

  signUp: {
    marginTop: 24,
  },
});

export default connect<DispatchProps>(
  mapStateToProps,
  mapDispatchToProps,
)(Login);
