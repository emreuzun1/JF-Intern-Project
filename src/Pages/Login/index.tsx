import React from 'react';
import {StackNavigationProp} from '@react-navigation/stack';
import {Text, Dimensions} from 'react-native';
import {connect} from 'react-redux';
import {RouteProp} from '@react-navigation/native';
import styled from 'styled-components/native';

import {RootStackParamList} from '../../Navigation/types';
import {requestLogin, resetForms} from '../../redux/actions';
import {Formik} from 'formik';
import {Colors} from '../../constants/Colors';

const logoImg = require('../../img/jotform-logo.png');
const {width} = Dimensions.get('screen');

type LoginProps = StackNavigationProp<RootStackParamList, 'Login'>;
type LoginRouteProps = RouteProp<RootStackParamList, 'Login'>;

const StyledContainer = styled.View({
  flex: 1,
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: Colors.jotformOrange,
});

const StyledTopContainer = styled.View({
  flex: 2,
  justifyContent: 'center',
});

const StyledLogoImage = styled.Image({
  width: width / 1.2,
  resizeMode: 'contain',
  marginBottom: 12,
});

const StyledBottomContainer = styled.View({
  flex: 3,
  backgroundColor: 'white',
  width: '100%',
  alignItems: 'center',
  borderTopLeftRadius: 36,
  borderTopRightRadius: 36,
  paddingTop: 24,
});

const StyledInputContainer = styled.View({
  alignItems: 'center',
  marginTop: 12,
});

const StyledTextInput = styled.TextInput({
  backgroundColor: Colors.lightGrey,
  borderRadius: 5,
  width: width / 1.6,
  height: 50,
  marginVertical: 10,
  fontSize: 16,
});

const StyledLoginButton = styled.TouchableOpacity({
  backgroundColor: Colors.jotformGrey,
  width: width / 2.6,
  padding: 12,
  alignItems: 'center',
  justifyContent: 'center',
  borderRadius: 8,
  marginTop: 24,
});

const StyledLoginText = styled.Text({
  color: 'white',
  fontSize: 18,
});

const StyledSignUpButton = styled.TouchableOpacity({
  marginTop: 24,
});

const StyledSignUpText = styled.Text({
  color: '#fa8900',
  fontWeight: 700,
});

interface Props {
  navigation: LoginProps;
  route: LoginRouteProps;
  requestLogin: (username: string, password: string) => void;
  resetForms: () => void;
}
interface IFormValues {
  username: string;
  password: string;
}

const Login = (props: Props) => {
  const initialValues: IFormValues = {username: '', password: ''};
  // eslint-disable-next-line no-shadow
  const {navigation, route, resetForms} = props;

  React.useEffect(() => {
    resetForms();
    if (route.params.isLogged) {
      navigation.navigate('Form');
    }
  }, [route.params.isLogged, navigation, resetForms]);

  return (
    <StyledContainer>
      <StyledTopContainer>
        <StyledLogoImage source={logoImg} />
      </StyledTopContainer>
      <StyledBottomContainer>
        <Formik
          initialValues={initialValues}
          onSubmit={values =>
            props.requestLogin(values.username, values.password)
          }>
          {({handleChange, handleBlur, handleSubmit, values}) => (
            <StyledInputContainer>
              <StyledTextInput
                autoCapitalize="none"
                placeholder=" Username"
                value={values.username}
                onBlur={handleBlur('username')}
                onChangeText={handleChange('username')}
              />
              <StyledTextInput
                autoCapitalize="none"
                placeholder=" Password"
                value={values.password}
                secureTextEntry
                onBlur={handleBlur('password')}
                onChangeText={handleChange('password')}
              />
              <StyledLoginButton onPress={handleSubmit}>
                <StyledLoginText>Login</StyledLoginText>
              </StyledLoginButton>
            </StyledInputContainer>
          )}
        </Formik>
        <StyledSignUpButton onPress={() => navigation.navigate('SignUp')}>
          <Text>
            Don't you have an account?
            <StyledSignUpText> Sign up!</StyledSignUpText>
          </Text>
        </StyledSignUpButton>
      </StyledBottomContainer>
    </StyledContainer>
  );
};

const mapStateToProps = (state: any) => {
  const {appKey} = state.auth;
  return {appKey};
};

const mapDispatchToProps = {
  requestLogin,
  resetForms,
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
