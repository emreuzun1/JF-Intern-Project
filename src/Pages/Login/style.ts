import {StyleSheet, Dimensions} from 'react-native';
import {Colors} from '../../constants/Colors';

const {width} = Dimensions.get('screen');

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.jotformOrange,
  },

  button: {
    backgroundColor: Colors.jotformGrey,
    width: width / 2.6,
    padding: 12,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 8,
    marginTop: 24,
  },

  loginText: {
    color: 'white',
    fontSize: 18,
    fontFamily: 'sf-regular',
  },

  topContainer: {
    flex: 2,
    justifyContent: 'center',
  },

  bottomContainer: {
    flex: 3,
    backgroundColor: 'white',
    width: '100%',
    alignItems: 'center',
    borderTopLeftRadius: 36,
    borderTopRightRadius: 36,
    paddingTop: 24,
  },

  inputContainer: {
    alignItems: 'center',
    marginTop: 12,
  },

  logoImg: {
    width: width / 1.2,
    resizeMode: 'contain',
    marginBottom: 12,
  },

  input: {
    backgroundColor: Colors.lightGrey,
    borderRadius: 5,
    width: width / 1.6,
    marginVertical: 10,
    fontSize: 16,
  },

  signUp: {
    marginTop: 24,
  },
});
