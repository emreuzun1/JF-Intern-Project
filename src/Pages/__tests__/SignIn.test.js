import React from 'react';
import {fireEvent, render, waitFor} from '../__mocks__/testUtils';
import Login from '../Login';

test('renders correctly', async () => {
  const mockedParams = {
    route: {params: {isLogged: false}},
    navigation: '',
  };
  const screen = render(<Login {...mockedParams} />);
  const usernameInput = screen.getByPlaceholderText(' Username');
  fireEvent.changeText(usernameInput, 'emreuzun1');
  const passwordInput = screen.getByPlaceholderText(' Password');
  fireEvent.changeText(passwordInput, 'w8h2h4lvjj.');
  const loginButton = screen.getByText('Login');
  fireEvent.press(loginButton);
});
