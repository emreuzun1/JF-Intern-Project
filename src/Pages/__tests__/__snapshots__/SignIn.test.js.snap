// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fa8900",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 2,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/img/jotform-logo.png",
        }
      }
      style={
        Array [
          Object {
            "marginBottom": 12,
            "resizeMode": "contain",
            "width": 625,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderTopLeftRadius": 36,
          "borderTopRightRadius": 36,
          "flexBasis": 0,
          "flexGrow": 3,
          "flexShrink": 1,
          "paddingTop": 24,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "marginTop": 12,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder=" Username"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "backgroundColor": "#EEEEEE",
              "borderRadius": 5,
              "fontSize": 16,
              "height": 50,
              "marginVertical": 10,
              "width": 468.75,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value="emreuzun1"
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder=" Password"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Array [
            Object {
              "backgroundColor": "#EEEEEE",
              "borderRadius": 5,
              "fontSize": 16,
              "height": 50,
              "marginVertical": 10,
              "width": 468.75,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value="w8h2h4lvjj."
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#434343",
            "borderRadius": 8,
            "justifyContent": "center",
            "marginTop": 24,
            "opacity": 1,
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
            "width": 288.46153846153845,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginTop": 24,
          "opacity": 1,
        }
      }
    >
      <Text>
        Don't you have an account?
        <Text
          style={
            Array [
              Object {
                "color": "#fa8900",
                "fontWeight": "700",
              },
            ]
          }
        >
           Sign up!
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
